### PR TITLE
[Mellanox] Add support for SN4280 platform

### PR DESCRIPTION
**Pull request by vivekrnv**
#### Why I did it

Add support for SN4280 platform including new components as cpld and fpga in the platform support part.
Add default and specific SKU.

#### How I did it
Added all relevant files to support new device and the relevant files to support a specific sku.

#### How to verify it
Verified by running regression on SN4280 platform. 

#### Which release branch to backport (provide reason below if selected)

- [ ] 202205
- [ ] 202305
- [ ] 202311
- [X] 202405

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

**Review by oleksandrivantsiv — APPROVED** [review]

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/buffers_defaults_t1.j2 [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+{% set default_cable = '300m' %}
+{%-set ports2cable = {
+            'spinerouter_leafrouter' : '2000m',
+            'leafrouter_torrouter' : '300m'
+       }
+-%}
+{% set ingress_lossless_pool_size =  '46743552' %}
+{% set ingress_lossless_pool_xoff  =  '8781696' %}
+{% set egress_lossless_pool_size =  '60817392' %}
+{% set egress_lossy_pool_size =  '46743552' %}
```
Comment (kperumalbfn) [inline device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/buffers_defaults_t1.j2:25]: Why different pool size and xoff values for T0 and T1? Any additional reserved buffers other than the number of ports and cable-length?

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/port_config.ini [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Ethernet176     176,177,178,179,180,181,182,183     etp23     23
+Ethernet184     184,185,186,187,188,189,190,191     etp24     24
+Ethernet192     192,193,194,195,196,197,198,199     etp25     25
+Ethernet200     200,201,202,203,204,205,206,207     etp26     26
+Ethernet208     208,209,210,211,212,213,214,215     etp27     27
+Ethernet216     216,217,218,219,220,221,222,223     etp28     28
+Ethernet224     224,225,226,227,228,229,230,231     etp29     29
+Ethernet232     232,233,234,235,236,237,238,239     etp30     30
+Ethernet240     240,241,242,243,244,245,246,247     etp31     31
+Ethernet248     248,249,250,251,252,253,254,255     etp32     32
```
Comment (kperumalbfn) [inline device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/port_config.ini:51]: What are the internal ports in this config? Could you add a comment here.

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/hwsku.json [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            "subport": "1"
+        },
+        "Ethernet216": {
+            "default_brkout_mode": "1x400G[200G,100G,50G,40G,25G,10G,1G]",
+            "subport": "1"
+        },
+        "Ethernet224": {
+            "default_brkout_mode": "1x400G",
+            "subport": "1",
+            "autoneg": "on",
```
Comment (kperumalbfn) [inline device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/hwsku.json:118]: Do we need autoneg on for these internal ports?

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/hwsku.json [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            "role": "Dpc"
+        },
+        "Ethernet248": {
+            "default_brkout_mode": "1x400G",
+            "subport": "1",
+            "autoneg": "on",
+            "role": "Dpc"
```
Comment (kperumalbfn) [inline device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/hwsku.json:137]: Is DPC == internal port?

Comment (vivekrnv) [reply device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/hwsku.json:137]: yes, `Dpc` is an indication for the internal port

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/hwsku.json (hunk 3 above)
Comment (vivekrnv) [reply device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/hwsku.json:118]: we do, autoneg is also enabled on the corresponding DPU port.

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/port_config.ini (hunk 2 above)
Comment (vivekrnv) [reply device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/port_config.ini:51]: This file is not used, latest config is derived from hwsku.json and platform.json, as long as they have Dpc config it's fine.

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/buffers_defaults_t1.j2 (hunk 1 above)
Comment (vivekrnv) [reply device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/buffers_defaults_t1.j2:25]: T1-T2: 2000m
T0-T1: 300m
T0-server: 40m

Cable lengths for uplink and downlink is different for T1 and T0 and so the variance in the values.

**Review by kperumalbfn — APPROVED** [review]
LGTM

**Comment by vivekrnv** [comment]
@liat-grozovik Can we merge this?

**Comment by yxieca** [comment]
@vivekrnv  please create 202311 PR manually. The auto-cherry-pick PR encountered build failures.

**Comment by vivekrnv** [comment]
@yxieca, Cherry-pick to 202311 is not required for this PR. It was a mistake to add the cherry-pick label. You can close the other PR. Sorry for the inconvinience

**Comment by bingwang-ms** [comment]
Hi @vivekrnv, the PR is required for PR https://github.com/sonic-net/sonic-buildimage/pull/19822 as per @ayurkiv-nvda. Can you please double confirm?

**Comment by vivekrnv** [comment]
> Hi @vivekrnv, the PR is required for PR #19822 as per @ayurkiv-nvda. Can you please double confirm?

Yes, checked with him. Cherry-pick is not required

**Pull request merged**